CONTRIBUTING.md

```diff
@@ -0,0 +1,290 @@
+- `tests/foo-bar` when the change concerns only the test suite
+- `refactor/foo-bar` when refactoring code without any behavior change
+- `style/foo-bar` when addressing some style issue
+- `docs/foo-bar` for updates to the README.md, this file, or similar documents
+
+[Back to TOC](#table-of-contents)
+
+#### Commit atomicity
+
+When submitting patches, it is important that you organize your commits in
+logical units of work. You are free to propose a patch with one or many
+commits, as long as their atomicity is respected. This means that no unrelated
+changes should be included in a commit.
+
+For example: you are writing a patch to fix a bug, but in your endeavour, you
+spot another bug. **Do not fix both bugs in the same commit!** Finish your
+work on the initial bug, propose your patch, and come back to the second bug
+later on. This is also valid for unrelated style fixes, refactors, etc...
+
+You should use your best judgment when facing such decisions. A good approach
+for this is to put yourself in the shoes of the person who will review your
+patch: will they understand your changes and reasoning just by reading your
+commit history? Will they find unrelated changes in a particular commit? They
+shouldn't!
+
+Writing meaningful commit messages that follow our commit message format will
+also help you respect this mantra (see the below section).
+
+[Back to TOC](#table-of-contents)
+
+#### Commit message format
+
+To maintain a healthy Git history, we ask of you that you write your commit
+messages as follows:
+
+- The tense of your message must be **present**
+- Your message must be prefixed by a type, and a scope
+- The header of your message should not be longer than 50 characters
+- A blank line should be included between the header and the body
+- The body of your message should not contain lines longer than 72 characters
+
+Here is a template of what your commit message should look like:
+
+```
+<type>(<scope>) <subject>
+<BLANK LINE>
+<body>
+<BLANK LINE>
+<footer>
+```
+
+##### Type
+
+The type of your commit indicates what type of change this commit is about. The
+accepted types are:
+
+- **feat**: A new feature
+- **fix**: A bug fix
+- **hotfix**: An urgent bug fix during a release process
+- **tests**: A change that is purely related to the test suite only (fixing
+  a test, adding a test, improving its reliability, etc...)
+- **docs**: Changes to the README.md, this file, or other such documents
+- **style**: Changes that do not affect the meaning of the code (white-space
+  trimming, formatting, etc...)
+- **perf**: A code change that significantly improves performance
+- **refactor**: A code change that neither fixes a bug nor adds a feature, and
+  is too big to be considered just `perf`
+- **chore**: Maintenance changes related to code cleaning that isn't
+  considered part of a refactor, build process updates, dependency bumps, or
+  auxiliary tools and libraries updates (LuaRocks, Travis-ci, etc...).
```

Comment:
We neither use LuaRocks nor Travis-ci here.

Reply:
artifacts of the kong doc, removed